src/Servant/Server/Internal/RoutingApplication.hs

```diff
@@ -49,36 +52,8 @@ data RouteResult a =
 toApplication :: forall m. MonadSnap m => RoutingApplication m -> Application m
 toApplication ra request respond = do
 
-  snapReq  <- getRequest
-  r        <- ra (request `addHeaders` headers snapReq) routingRespond
-  rspnd <- respond r
-
-  -- liftIO $ putStrLn $ unlines [
-  --   "----------"
-  --   , "SNAP REQ"
-  --   , show snapReq
-  --   , "----------"
-  --   , "request"
-  --   , show request
-  --   , "----------"
-  --   , "r"
-  --   , show r
-  --   , "----------"
-  --   , "snapResp"
-  --   , show snapResp
-  --   , "----------"
-  --   , "rspnd"
-  --   , show rspnd
-  --   ]
-
-  return rspnd
-
-  -- snapReq  <- getRequest
-  -- r        <- ra (request `addHeaders` headers snapReq) routingRespond
```

Comment:
This is the only important part of the PR - removing the addition of `snapReq`'s headers to `request`. These headers are _already_ added by `snapToApplication`.